src/java_bytecode/generate_java_generic_type.cpp

```diff
@@ -224,24 +234,26 @@ irep_idt generate_java_generic_typet::build_generic_tag(
   return new_tag_buffer.str();
 }
 
-/// Build the specalised version of the specific class, with the specified
+/// Build the specialised version of the specific class, with the specified
 /// parameters and name.
 /// \param generic_class_definition: The generic class we are specialising
 /// \param new_tag: The new name for the class (like Generic<java::Float>)
 /// \param new_components: The specialised components
 /// \return The newly constructed class.
-java_class_typet
+java_specialised_generic_class_typet
 generate_java_generic_typet::construct_specialised_generic_type(
   const java_generic_class_typet &generic_class_definition,
   const irep_idt &new_tag,
   const struct_typet::componentst &new_components) const
 {
-  java_class_typet specialised_class = generic_class_definition;
+  java_specialised_generic_class_typet specialised_class{};
```

Comment:
Rather than constructing an empty `java_specialised_generic_class_typet` and then filling in it's fields with a bunch of `set`s, would it be cleaner/safer to actually add a proper constructor to the class type?